freqtrade/data/converter/converter.py

```diff
@@ -44,6 +48,285 @@ def ohlcv_to_dataframe(ohlcv: list, timeframe: str, pair: str, *,
+    df['price'] = ((trades['price'] / scale).round()
+                   * scale).astype('float64').values
+    if df.empty:
+        df['total'] = np.nan
+        df['delta'] = np.nan
+        return df
+
+    df['delta'] = df['ask_amount'] - df['bid_amount']
+    df['total_volume'] = df['ask_amount'] + df['bid_amount']
+    df['total_trades'] = df['ask'] + df['bid']
+
+    # group to bins aka apply scale
+    df = df.groupby('price').sum(numeric_only=True)
+    return df
+
+
+def trades_orderflow_to_imbalances(df: DataFrame, imbalance_ratio: int, imbalance_volume: int):
+    """
+    :param df: dataframes with bid and ask
+    :param imbalance_ratio: imbalance_ratio e.g. 300
+    :param imbalance_volume: imbalance volume e.g. 3)
+    :return: dataframe with bid and ask imbalance
+    """
+    bid = df.bid
+    ask = df.ask.shift(-1)
+    bid_imbalance = (bid / ask) > (imbalance_ratio / 100)
+    # overwrite bid_imbalance with False if volume is not big enough
+    bid_imbalance_filtered = np.where(
+        df.total_volume < imbalance_volume, False, bid_imbalance)
+    ask_imbalance = (ask / bid) > (imbalance_ratio / 100)
+    # overwrite ask_imbalance with False if volume is not big enough
+    ask_imbalance_filtered = np.where(
+        df.total_volume < imbalance_volume, False, ask_imbalance)
+    dataframe = DataFrame(
+        {'bid_imbalance': bid_imbalance_filtered, 'ask_imbalance': ask_imbalance_filtered}, index=df.index)
+
+    return dataframe
+
+
+def stacked_imbalance(df: DataFrame, label: str = "bid", stacked_imbalance_range: int = 3, should_reverse: bool = False):
+    """
+    y * (y.groupby((y != y.shift()).cumsum()).cumcount() + 1)
+    https://stackoverflow.com/questions/27626542/counting-consecutive-positive-values-in-python-pandas-array
+    """
+    imbalance = df[f'{label}_imbalance']
+    int_series = pd.Series(np.where(imbalance, 1, 0))
+    stacked = int_series * \
+        (int_series.groupby((int_series != int_series.shift()).cumsum()).cumcount() + 1)
+
+    max_stacked_imbalance_idx = stacked.index[stacked >=
+                                              stacked_imbalance_range]
+    stacked_imbalance_price = np.nan
+    if not max_stacked_imbalance_idx.empty:
+        # TODO: do better than just take first
+        idx = max_stacked_imbalance_idx[0] if not should_reverse else np.flipud(
+            max_stacked_imbalance_idx)[0]
+        stacked_imbalance_price = imbalance.index[idx]
+    return stacked_imbalance_price
+
+
+def stacked_imbalance_bid(df: DataFrame, stacked_imbalance_range: int = 3):
+    return stacked_imbalance(df, 'bid', stacked_imbalance_range)
+
+
+def stacked_imbalance_ask(df: DataFrame, stacked_imbalance_range: int = 3):
+    return stacked_imbalance(df, 'ask', stacked_imbalance_range, should_reverse=True)
+
+
+def orderflow_to_volume_profile(orderflow: DataFrame):
+    """
+    :param orderflow: dataframe
+    :return: volume profile dataframe
+    """
+    df = orderflow
```

Comment:
You might want to either use the orderflow name - OR change the parameter to df.

taking an "orderflow" parameter - and then immediately aliasing it as something else (this line will NOT performa a copy) seems kinda pointless.